kolibri/plugins/management/assets/src/vue/manage-class-page/class-delete-modal.vue

```diff
@@ -68,14 +62,10 @@
         required: true,
       },
     },
-    data() {
-      return {
-        error_message: '',
-      };
-    },
     methods: {
-      submit() {
-        // need delete logic here.
+      classDelete() {
+        this.deleteClass(this.classid);
+        this.$emit('close');
```

Comment:
Two comments here:

1. I'd like to try to avoid using this event-driven pattern. If possible, I think it's cleaner to add additional page state to the vuex store and have that drive whether the modals are visible or not. See the import/export modals for an example of this in practice.
2. Currently, this logic will close the modal regardless of whether the operation fails. Moving to a vuex-driven strategy inside of the action's promise will make it easier to persist and display an error message rather than closing the modal.